File: easy/problems/0507-perfect-number/solution.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num <= 1:
            return False

        total = 1
        factor = 2
        while factor * factor <= num:
            if num % factor == 0:
                total += factor
                if factor * factor != num:
                    total += num // factor
            factor += 1
        return total == num

File: easy/problems/0507-perfect-number/solution.cpp
class Solution {
public:
    bool checkPerfectNumber(int num) {
        if (num <= 1) {
            return false;
        }

        int total = 1;
        for (int factor = 2; factor * factor <= num; factor++) {
            if (num % factor == 0) {
                total += factor;
                if (factor * factor != num) {
                    total += num / factor;
                }
            }
        }
        return total == num;
    }
};
